include/cudaq/Support/CMakeLists.txt

```
@@ -0,0 +1,13 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+execute_process(COMMAND git describe --tags --abbrev=0 --dirty=-developer
```

I believe this will currently show `0.4.0-alpha`, which seems a little bit misleading because not only is this not a 0.4.0 alpha release, we're well past the `0.4.0` release (which is even further past the alpha).
What do you think about something like this (which I haven't fully tried in a CMake context, so it may need to be adjusted):
```suggestion
execute_process(COMMAND git describe --tags --exact-match 2> /dev/null || git describe --tags --long --dirty=-developer
```
Which will show something like this: `0.4.0-alpha-214-g03cc85df`, which means 214 commits after the 0.4.0-alpha tag?

```bash
$ git describe --tags --exact-match 2> /dev/null || git describe --tags --long --dirty=-developer
0.4.0-alpha-214-g03cc85df
$ git checkout 0.4.0
$ git describe --tags --exact-match 2> /dev/null || git describe --tags --long --dirty=-developer
0.4.0
```

---

I decided on the given format because it sort of follows that of LLVM. When you build clang on the main branch you'll get a "version" that is the next release. Ideally, we'd push a label like 'proto-0.5.0' to our main branch. I didn't do that, so yeah, it's a bit squirrely at the moment.

---

Actually, I don't know that I have permissions to even push tags. I haven't tried.